src/components/MovieForm/MovieForm.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '../Dialog/Dialog';
+import { Box, Input, InputLabel, TextField } from '@mui/material';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import dayjs from 'dayjs';
+import { DatePicker } from '@mui/x-date-pickers';
+import enGb from 'dayjs/locale/en-gb';
+
+const MovieForm = ({ info, genreList, onSubmit, onClose, isVisible }) => {
+  // QUESTION: would it be better to use useReducer here? or it will be over-complication?
+  const [movieInfo, setMovieInfo] = useState({
+    ...info,
+    genre: info.genre ? [...info.genre] : [],
+    releaseDate: info.releaseDate ? dayjs(info.releaseDate, 'YYYY-MM-DD') : null,
+  });
+
+  const isEditDialog = !!movieInfo.id || false;
```

Comment:
should be enough to get `true` or `false`
```suggestion
  const isEditDialog = !!movieInfo.id;
```